### PR TITLE
Source page update

### PR DESCRIPTION
**Pull request by kmshin1397**
source

- Update Source page to use MUI Accordions and Papers
- Make page responsive by adding a separate SourceMobile component for smaller screens to allow for distinct layout differences between desktop/mobile
- Various style changes to Source page components to make things fit on mobile (i.e. overflows)
- Propagate changes to Candidate page
- On smaller screens, Comments are replaced with a CommentsMobile component which only displays the most recent sources and adds a button for a dialog to see all comments/add a new comment

### CODE REVIEW

**Comment by dmitryduev** [comment]
Awesome @kmshin1397 !! Could you please pin the latest baselayer here?

File: static/js/components/AssignmentList.jsx [hunk 1]
```diff
@@ -93,14 +113,14 @@ const AssignmentList = ({ assignments }) => {
                 <td>{assignment.comment}</td>
                 <td>
                   <span>
-                    <button
-                      type="button"
+                    <IconButton
+                      aria-label="delete"
```
Comment (profjsb) [inline static/js/components/AssignmentList.jsx:117]: make this label more descriptive e.g. "delete-assignment"

File: static/js/components/Candidate.jsx [hunk 2]
```diff
@@ -45,75 +69,111 @@ const Candidate = ({ route }) => {
...
+              <div>
+                <b>Position (J2000):</b>
+                &nbsp;
+                {candidate.ra}, &nbsp;
+                {candidate.dec}
+                &nbsp; (&alpha;,&delta;=
+                {ra_to_hours(candidate.ra)}, &nbsp;
+                {dec_to_hours(candidate.dec)}) &nbsp; (l,b=
+                {candidate.gal_lon.toFixed(1)}, &nbsp;
+                {candidate.gal_lat.toFixed(1)}
```
Comment (profjsb) [inline static/js/components/Candidate.jsx:92]: I think this is `toFixed(5)` in Sources. Right @guynir42 @dannygoldstein?

File: static/js/components/SourceDesktop.jsx [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+          />
+          <b>Position (J2000):</b>
+          &nbsp;
+          {source.ra}, &nbsp;
+          {source.dec}
+          &nbsp; (&alpha;,&delta;=
+          {ra_to_hours(source.ra)}, &nbsp;
+          {dec_to_hours(source.dec)}) &nbsp; (l,b=
+          {source.gal_lon.toFixed(1)}, &nbsp;
+          {source.gal_lat.toFixed(1)}
```
Comment (profjsb) [inline static/js/components/SourceDesktop.jsx:142]: Hmm. See comment above.

**Comment by kmshin1397** [comment]
> Awesome @kmshin1397 !! Could you please pin the latest baselayer here?

I merged in the latest master last night so I believe the baselayer is already at the latest commit.

**Review by profjsb — COMMENTED** [review]
Looks great. Few questions inline.

**Comment by dmitryduev** [comment]
@stefanv merged the latest stuff in BL a few hours ago, so need a refresh

File: static/js/components/Candidate.jsx (hunk 2 above)
Comment (guynir42) [reply static/js/components/Candidate.jsx:92]: toFixed(6)

**Comment by kmshin1397** [comment]
#792

**Review by profjsb — CHANGES_REQUESTED** [review]
Looks great. When loading on a small screen I get one console error:
```
Warning: Failed prop type: Invalid prop `source.loadError` of type `string` supplied to `SourceMobile`, expected `boolean`.
    in SourceMobile (created by Source)
    in Source (created by Context.Consumer)
    in ErrorBoundary (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in Route (created by PropsRoute)
    in PropsRoute (created by MainContent)
    in Switch (created by MainContent)
    in div (created by MainContent)
    in MuiPickersUtilsProvider (created by MainContent)
    in ThemeProvider (created by Theme)
    in Theme (created by MainContent)
    in MainContent
    in Router (created by BrowserRouter)
    in BrowserRouter
    in ErrorBoundary (created by ConnectFunction)
    in ConnectFunction
    in Provider
```
Also, while the photometry and the spectroscopy plots are not responsive, can the offset plot be made responsive?

**Comment by kmshin1397** [comment]
> Looks great. When loading on a small screen I get one console error:
> 
> ```
> Warning: Failed prop type: Invalid prop `source.loadError` of type `string` supplied to `SourceMobile`, expected `boolean`.
>     in SourceMobile (created by Source)
>     in Source (created by Context.Consumer)
>     in ErrorBoundary (created by ConnectFunction)
>     in ConnectFunction (created by Context.Consumer)
>     in Route (created by PropsRoute)
>     in PropsRoute (created by MainContent)
>     in Switch (created by MainContent)
>     in div (created by MainContent)
>     in MuiPickersUtilsProvider (created by MainContent)
>     in ThemeProvider (created by Theme)
>     in Theme (created by MainContent)
>     in MainContent
>     in Router (created by BrowserRouter)
>     in BrowserRouter
>     in ErrorBoundary (created by ConnectFunction)
>     in ConnectFunction
>     in Provider
> ```
> 
> Also, while the photometry and the spectroscopy plots are not responsive, can the offset plot be made responsive?

The error should be fixed now. I'll explore the offset plot responsiveness but I suspect it may not be straightforward. I know @dmitryduev tried for awhile to get the Vega plots on his Alerts stuff do be responsive and couldn't get it to cooperate.

**Comment by kmshin1397** [comment]
@profjsb Got the offset plot to be responsive.

File: static/js/components/Source.jsx [hunk 4]
```diff
@@ -80,154 +58,18 @@ const Source = ({ route }) => {
...
-          title="Centroid Plot"
-          mobileProps={{ folded: true }}
-        >
-          <Suspense fallback={<div>Loading centroid plot...</div>}>
-            <CentroidPlot className={styles.smallPlot} sourceId={source.id} />
-          </Suspense>
-        </Responsive>
-      </div>
-    </div>
+  return (
```
Comment (acrellin) [inline static/js/components/Source.jsx:69]: Any reason not to combine these into a single return statement with something like 
``` javascript
{width <= 1200 ? <SourceMobile source={source} /> : <SourceDesktop source={source} />}
```
?

File: static/js/components/CommentListMobile.jsx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    ({
+      id,
+      author,
+      author_info,
+      created_at,
+      text,
+      attachment_name,
+      groups,
+    }) => {
+      return (
```
Comment (acrellin) [inline static/js/components/CommentListMobile.jsx:134]:
```suggestion
    }) => (
```

File: static/js/components/CommentListMobile.jsx [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+                  <a href={`/api/comment/${id}/attachment`}>
+                    {attachment_name}
+                  </a>
+                </div>
+              )}
+            </span>
+          </div>
+        </span>
+      );
+    }
```
Comment (acrellin) [inline static/js/components/CommentListMobile.jsx:205]:
```suggestion
      )
```

File: static/js/components/CommentListMobile.jsx (hunk 5 above)
Comment (acrellin) [reply static/js/components/CommentListMobile.jsx:134]: (with associated indentation changes following, though `prettier` will take care of that now :wink: )

File: static/js/components/CommentListMobile.jsx [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+  let { comments } = obj;
+
+  comments = comments || [];
+
+  const emojiSupport = (text) =>
+    text.value.replace(/:\w+:/gi, (name) => emoji.getUnicode(name));
+
+  const recentComments = comments.slice(0, 3);
+
+  const items = recentComments.map(
```
Comment (acrellin) [inline static/js/components/CommentListMobile.jsx:124]: I think we should either factor this out into a real React component, or just render it inline below in the main return statement and not assign it here.

Comment (acrellin) [reply static/js/components/CommentListMobile.jsx:124]: I think just moving this map inline below in the main return statement would be best

**Review by acrellin — APPROVED** [review]
Looks really great, thanks @kmshin1397 ! Just a few minor comments below

**Review by profjsb — APPROVED** [review]

**Pull request merged**